### PR TITLE
Enhance documentation about usage of `value` and `gas`

### PR DESCRIPTION
**Pull request by Marenz**
People might look for information in the members section of the function types chapter.

### CODE REVIEW

File: docs/control-structures.rst [hunk 1]
```diff
@@ -91,7 +91,7 @@ You need to use the modifier ``payable`` with the ``info`` function because
 otherwise, the ``.value()`` option would not be available.
 
 .. warning::
-  Be careful that ``feed.info.value(10).gas(800)`` only locally sets the ``value`` and amount of ``gas`` sent with the function call, and the parentheses at the end perform the actual call. So in this case, the function is not called.
+  Be careful that ``feed.info.value(10).gas(800)`` only locally sets the ``value`` and amount of ``gas`` sent with the function call, and the parentheses at the end perform the actual call. So in this case, the function is not called and the ``value`` and ``gas`` values are lost.
```
Comment (chriseth) [inline docs/control-structures.rst]:
```suggestion
  Be careful that ``feed.info.value(10).gas(800)`` only locally sets the ``value`` and amount of ``gas`` sent with the function call, and the parentheses at the end perform the actual call. So in this case, the function is not called and the ``value`` and ``gas`` settings are lost.
```

File: docs/types/value-types.rst [hunk 2]
```diff
@@ -625,6 +625,9 @@ which returns the :ref:`ABI function selector <abi_function_selector>`::
       }
     }
 
+.. note::
+    Public (or external) payable functions also have the pseudo-members ``value`` and ``gas`` but those are only to be used in combination with a call to the function, see :ref:`External Function Calls <external-function-calls>` for more information.
```
Comment (chriseth) [inline docs/types/value-types.rst]: I would say `value` and `gas` are not more pseudo than `selector` and we should properly document them here.

**Comment by Marenz** [comment]
Updated

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Test fails.

**Comment by Marenz** [comment]
Test succeeds now

**Review by chriseth — APPROVED** [review]

**Pull request merged**